lib/node/runner/checker_ballot_transaction.go

```diff
@@ -282,14 +288,71 @@ func ValidateOp(st *storage.LevelDBBackend, source *block.BlockAccount, op trans
 			err = errors.ErrorFrozenAccountNoDeposit
 			return
 		}
-		// If it's a frozen account, everything must be withdrawn
 		if source.Linked != "" {
+			// If it's a frozen account, everything must be withdrawn
 			var expected common.Amount
 			expected, err = source.Balance.Sub(common.BaseFee)
 			if casted.Amount != expected {
 				err = errors.ErrorFrozenAccountMustWithdrawEverything
 				return
 			}
+			// Unfreezing must be done after X period from unfreezing request
+			iterFunc, closeFunc := block.GetBlockOperationsBySource(st, source.Address, nil)
+			bo, _, _ := iterFunc()
+			closeFunc()
+
+			var lastblock block.Block
+			lastblock, err = block.GetLatestBlock(st)
+			if err != nil {
+				return
+			}
+			if bo.Type != transaction.OperationUnfreezingRequest {
+				err = errors.ErrorUnfreezingRequestNotRequested
+				return
+			}
+
+			// unfreezing period is 241920.
+			if lastblock.Height-bo.BlockHeight <= common.UnfreezingPeriod {
+				err = errors.ErrorUnfreezingNotReachedExpiration
+				return
+			}
+		}
+	case transaction.OperationUnfreezingRequest:
```

Comment:
The original design (which we discussed) was that the "unfreezing request" would just be a payment operation.
That will allow you to reuse most of this function with much less code.

Reply:
@Geod24 Do you mean if some user wants unfreezing, he only need to send two payment txs at X blockheights interval?